tasks/image-segmentation/src/validation.py

```diff
@@ -0,0 +1,184 @@
+        return 0.0
+
+    for label in classLabels:
+        if label > 0:
+            testMaskClass = testMask == label
+            predictedMaskClass = predictedMask == label
+            iouScores.append(iouScoreClass(testMaskClass, predictedMaskClass))
+
+    return sum(iouScores) / len(iouScores)
+
+
+def iouScoreImages(testMasks: list[np.ndarray], predictedMasks: list[np.ndarray], samples: list[ImageSample]) -> list[float]:
+    iouScores: list[float] = []
+    for testMask, predictedMask, sample in zip(testMasks, predictedMasks, samples):
+        iou = iouScoreImage(testMask, predictedMask)
+        iouScores.append(iou)
+        logging.info(f">> [Image Segmentation] IoU Score for sample {sample.name} (sample id: {sample.id}) is: {round(iou, 2)}")
+
+    return iouScores
+
+
+def addPadding(predictedMasks: list[np.ndarray], testMasksPadding: list[tuple[int, int]]) -> list[np.ndarray]:
+    for index, mask in enumerate(predictedMasks):
+        vPadding = testMasksPadding[index][0]
+        hPadding = testMasksPadding[index][1]
+
+        if vPadding > 0:
+            mask[:vPadding, :] = 0
+            mask[-vPadding:, :] = 0
+        if hPadding > 0:
+            mask[:, :hPadding] = 0
+            mask[:, -hPadding:] = 0
+
+    return predictedMasks
+
+
+def run(model: tf.lite.Interpreter, batchImages: np.ndarray) -> np.ndarray:
+    inputDetails = model.get_input_details()
+    outputDetails = model.get_output_details()
+
+    model.resize_tensor_input(inputDetails[0]["index"], batchImages.shape)
+    model.allocate_tensors()
+    model.set_tensor(inputDetails[0]["index"], batchImages.astype(np.float32))
+    model.invoke()
+
+    prediction = model.get_tensor(outputDetails[0]["index"])
+    prediction = np.argmax(prediction, axis = -1)
+
+    return prediction
+
+
+def predictionPrepare(batchSamples: list[ImageSample], imageSize: int, classes: ImageDatasetClasses) -> tuple[np.ndarray, list[np.ndarray], list[tuple[int, int]]]:
+    testImgs: list[np.ndarray] = []
+    testMasks: list[np.ndarray] = []
+    testMasksPadding: list[tuple[int, int]] = []
+
+    for sample in batchSamples:
+        sample.download()
+        sample.unzip()
+        sampleData = sample.load()
+
+        if hasDotAnnotation(sampleData.annotation):
+            logging.warning(f">> [Image Segmentation] Sample \"{sample.name}\" (ID: {sample.id}) has invalid annotation (too few coordinates). Skipping Sample")
+            continue
+
+        resized, _, _ = resizeWithPadding(sampleData.image, imageSize, imageSize)
+        normalized = resized / 255
+
+        groundtruth = sampleData.extractSegmentationMask(classes)
+        groundtruth, vPadding, hPadding = resizeWithPadding(groundtruth, imageSize, imageSize)
+
+        testImgs.append(normalized)
+        testMasks.append(groundtruth)
+        testMasksPadding.append((vPadding, hPadding))
+
+    testImages = np.array(testImgs)
+
+    return (testImages, testMasks, testMasksPadding)
```

Comment:
Brackets are not needed when returning multiple values
`return variable1, variable2` works fine